### PR TITLE
[SPARK-42305][SQL] Integrate `_LEGACY_ERROR_TEMP_1229` into `DECIMAL_PRECISION_EXCEEDS_MAX_PRECISION`

### PR DESCRIPTION
**Pull request by itholic**
### What changes were proposed in this pull request?

This PR proposes to integrate `_LEGACY_ERROR_TEMP_1229` into `DECIMAL_PRECISION_EXCEEDS_MAX_PRECISION`.

**_LEGACY_ERROR_TEMP_1229**
```json
  "_LEGACY_ERROR_TEMP_1229" : {
    "message" : [
      "<decimalType> can only support precision up to <precision>."
    ]
  },
```

**DECIMAL_PRECISION_EXCEEDS_MAX_PRECISION**
```json
  "DECIMAL_PRECISION_EXCEEDS_MAX_PRECISION" : {
    "message" : [
      "Decimal precision <precision> exceeds max precision <maxPrecision>."
    ],
    "sqlState" : "22003"
  },
```

### Why are the changes needed?

We should assign proper name to _LEGACY_ERROR_TEMP_*

### Does this PR introduce _any_ user-facing change?

No

### How was this patch tested?

`./build/sbt "sql/testOnly org.apache.spark.sql.SQLQueryTestSuite*"`

### CODE REVIEW

**Comment by itholic** [comment]
I think we can reuse `DECIMAL_PRECISION_EXCEEDS_MAX_PRECISION ` for `_LEGACY_ERROR_TEMP_1229 `.

cc @srielau @MaxGekk @cloud-fan

**Review by srielau — APPROVED** [review]
LGTM

File: sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/parser/ExpressionParserSuite.scala [hunk 1]
```diff
@@ -804,14 +804,11 @@ class ExpressionParserSuite extends AnalysisTest {
       Literal(BigDecimal("90912830918230182310293801923652346786").underlying()))
     assertEqual("123.0E-28BD", Literal(BigDecimal("123.0E-28").underlying()))
     assertEqual("123.08BD", Literal(BigDecimal("123.08").underlying()))
+    val e = intercept[SparkArithmeticException](defaultParser.parseExpression("1.20E-38BD"))
     checkError(
-      exception = parseException("1.20E-38BD"),
-      errorClass = "_LEGACY_ERROR_TEMP_0061",
```
Comment (MaxGekk) [inline sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/parser/ExpressionParserSuite.scala:809]: Sorry, I didn't get. The PR title and description are telling only about one error class `LEGACY_ERROR_TEMP_1229` but your PR affects another one. Could you elaborate this and revert unrelated changes, or make PR's title/description consistent with PR's changes.

**Review by MaxGekk — CHANGES_REQUESTED** [review]

File: sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/parser/ExpressionParserSuite.scala (hunk 1 above)
Comment (itholic) [reply sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/parser/ExpressionParserSuite.scala:809]: Yeah, you're right.
I should have fixed `sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/AstBuilder.scala` rather than tests.
Just reverted unrelated changes and fix them in proper way.
Thanks!

**Review by LuciferYang — APPROVED** [review]
+1, LGTM

**Comment by MaxGekk** [comment]
+1, LGTM. Merging to master/3.4.
Thank you, @itholic and @srielau @LuciferYang for review.
